Two Sums - Unsorted/Two Sums - Unsorted.js

```diff
@@ -0,0 +1,57 @@
+/*
+
+Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
+
+You may assume that each input would have exactly one solution, and you may not use the same element twice.
+
+You can return the answer in any order.
+
+ 
+
+Example 1:
+
+Input: nums = [2,7,11,15], target = 9
+Output: [0,1]
+Explanation: Because nums[0] + nums[1] == 9, we return [0, 1].
+Example 2:
+
+Input: nums = [3,2,4], target = 6
+Output: [1,2]
+Example 3:
+
+Input: nums = [3,3], target = 6
+Output: [0,1]
+ 
+
+Constraints:
+
+2 <= nums.length <= 104
+-109 <= nums[i] <= 109
+-109 <= target <= 109
+Only one valid answer exists.
+ 
+
+Follow-up: Can you come up with an algorithm that is less than O(n2) time complexity?
+
+*/
+
+const fs = require("fs");
+const arrayNums = fs.readFileSync("data.txt", "utf-8").split(",");
+let desiredTarget = arrayNums.pop();
+
+function findThePair(input, target) {
+    let object = {};
+
+    for (let i = 0; i < input.length; i++) {
+        if (Number.isInteger(object[target - input[i]])) {

```

Comment:
It looks like object's in Javascript have a way of determining if they have a key/field set. I think you could do something like:
```
if (target - input[i] in object) {
```
Reads a little nicer to me, we're asking if a certain key exists `in` that object

Again, might not be the most idiomatic way of doing things in Javascript, @eric-decompiled can probably chime in on this too.

Reply:
This is an unfamiliar syntax to me.  I will await @eric-decompiled to pronounce judgement and make changes accordingly.

Reply:
I got the thumbs up ;)